src/enzyme_ad/jax/Passes/AffineToStableHLORaising.cpp

```diff
@@ -944,6 +978,223 @@ emitStoreAsScatter(Location loc, Value update, Value input, ValueRange sIndices,
   return res;
 }
 
+static LogicalResult tryRaisingForOpToStableHLOUnroll(
+    affine::AffineForOp forOp, IRMapping &mapping, OpBuilder &builder,
+    llvm::DenseMap<Value, affine::AffineValueMap> &maps, ParallelContext pc) {
+  // Materialize an unrolled version of the loop in a temporary block and
+  // generate the raised version of that. The unrolled version will be deleted
+  // afterwards and the results of the original for loop will be mapped to it.
+
+  // There arises a problem with the affine maps contained in the for loop which
+  // until now correctly identified the loop iv as an affine dim, but will now
+  // take constants as inputs. We need to canonicalize those maps before raising
+  // the operations because we assume dim inputs to be loop ivs. This is the
+  // reason we need to canonicalize all affine maps before we raise them in the
+  // other parts of the code.
```

Comment:
@ftynse I wanted to ask you about this part. We would want to canonicalize the affine exprs in the unrolled version of the loop, but as far as I know we can't run the pattern rewriter in just a block  because it needs an IsolatedFromAbove operation. Is there a nice way to be able to reuse the canonicalizers or would we need to rewrite/vendor them in?

We would like to do the unroll while in the raising sequence because we want the option to first try the for->while raising and only fallback to this if that doesn't succeed.
But perhaps it might be better to attempt the raising and if we fail due to an infeasible while->for unroll that specific loop, try again and repeat until we manage to raise.

Not sure if there is a nicer way to do this and I am overthinking it.